flutter/lib/util/playlist_creation.dart

```diff
@@ -0,0 +1,331 @@
+    return CreatePlaylist.fromJson(playlistMap!);
+  }
+
+  Future<UserProfile?> getUserProfile() async {
+    final url = Uri.https('api.spotify.com', 'v1/me');
+
+    final userProfileMap = await callSpotify(url);
+    return UserProfile.fromJson(userProfileMap!);
+  }
+
+  Future<AudioFeatures?> getAudioFeatures(String trackQuery) async {
+    final url = Uri.https(
+      'api.spotify.com',
+      '/v1/audio-features',
+      {'ids': trackQuery},
+    );
+
+    final audioFeaturesMap = await callSpotify(url);
+    setStatus(audioFeaturesMap.toString());
+    return AudioFeatures.fromJson(audioFeaturesMap!);
+  }
+
+  Future<TopTracks?> getTopTracks(String artistUri) async {
+    final url = Uri.https(
+      'api.spotify.com',
+      '/v1/artists/$artistUri/top-tracks',
+      {'market': 'US'},
+    );
+
+    final topTracksMap = await callSpotify(url);
+    return TopTracks.fromJson(topTracksMap!);
+  }
+
+  Future<FollowedArtists?> getFollowedArtists({required int limit}) async {
+    final url = Uri.https(
+      'api.spotify.com',
+      '/v1/me/following',
+      {'limit': limit.toString(), 'type': 'artist'},
+    );
+
+    final followedArtistsMap = await callSpotify(url);
+    return FollowedArtists.fromJson(followedArtistsMap!);
+  }
+
+  Future<TopArtists?> getTopArtists({
+    required int limit,
+    required String timeRange,
+  }) async {
+    final url = Uri.https(
+      'api.spotify.com',
+      '/v1/me/top/artists',
+      {'limit': limit.toString(), 'time_range': timeRange},
+    );
+
+    final topArtistsMap = await callSpotify(url);
+    return TopArtists.fromJson(topArtistsMap!);
+  }
+
+  Future<Map<String, dynamic>?> callSpotifyPost(
+    Uri url,
+    String body, {
+    bool refreshToken = false,
+  }) async {
+    final response = await http.post(
+      url,
+      headers: {'Authorization': await getAuthHeader(retry: refreshToken)},
+      body: body,
+    );
+
+    // Handle response
+    if (response.statusCode == 201) {
+      final jsonResponse =
+          convert.jsonDecode(response.body) as Map<String, dynamic>;
+      return jsonResponse;
+    } else if (response.statusCode == 401) {
+      return callSpotifyPost(url, body, refreshToken: true);
+    } else if (response.statusCode == 400) {
+      setStatus('Request failed with status: ${response.statusCode}.');
+      final jsonResponse =
+          convert.jsonDecode(response.body) as Map<String, dynamic>;
+      setStatus(jsonResponse.toString());
+    } else {
+      setStatus('Request failed with status: ${response.statusCode}.');
+      return null;
+    }
+    return null;
+  }
+
+  Future<Map<String, dynamic>?> callSpotify(
+    Uri url, {
+    bool refreshToken = false,
+  }) async {
+    final response = await http.get(
+      url,
+      headers: {'Authorization': await getAuthHeader(retry: refreshToken)},
+    );
+
+    // Handle response
+    if (response.statusCode == 200 || response.statusCode == 201) {
+      final jsonResponse =
```

Comment:
Here is an extra variable, it is preferable to simply return.

Reply:
Changed.